test/framework/src/main/java/org/elasticsearch/test/rest/yaml/section/MatchAssertion.java

```diff
@@ -81,7 +90,36 @@ protected void doAssert(Object actualValue, Object expectedValue) {
             }
         }
 
-        if (expectedValue.equals(actualValue) == false) {
+        // add support for matching objects ({a:b}) against list of objects ([ {a:b, c:d} ])
+        if(expectedValue instanceof Map && actualValue instanceof List) {
+            Map<String, Object> expectedMap = (Map<String, Object>) expectedValue;
+            List<Object> actualList = (List<Object>) actualValue;
+            assertTrue(
```

Comment:
I'm not sure we want `match` to be `contains` in some cases. If we want this we should make a new construct.

Reply:
The semantics seemed to fit with me, but now that I think of it, `match` is in fact an exact match so it should be an exact match when dealing with lists as well. I'll add a new construct and call it `contains`.